test/mlang/type-alias.mc

```diff
@@ -0,0 +1,70 @@
+-- Miking is licensed under the MIT license.
+-- Copyright (C) David Broman. See file LICENSE.txt
+--
+-- Type-aliases defined within language fragments.
+
+lang L1
+  syn T1 =
+
+  type A1 = {a1 : T1, a2 : T2}
+  type A2 = {b1 : A1, b2 : Int}
+
+  syn T2 =
```

Comment:
I'd suggest also adding `syn`s using `type`s that are defined later, since that's allowed by the semantics we're aiming for. After that I think this looks good.